web/js/barchartview.js

```diff
@@ -0,0 +1,187 @@
+class BarChartView extends View {
+
+    constructor(data, svg, name, size, margin, option) {
+        super(data, svg, size);
+        this.name = name
+        this.margin = margin;
+        this.container_width = size.width;
+        this.container_height = size.height;
+        this.content_width = this.container_width -this.margin.left -this.margin.right;
+        this.content_height = this.container_height -this.margin.top -this.margin.bottom;
+        this.svg
+            .attr('class', this.name+'_svg')
+            .attr('width', this.container_width)
+            .attr('height', this.container_height);
+        this.content_area = this.svg.append('g')
+            .attr('class', this.name+'_content_area')
+            .attr('transform', 'translate(' + this.margin.left + ',' + this.margin.top + ')')
+            .attr('width', this.content_width)
+            .attr('height', this.content_height);
+        this.xAxis = this.svg.append('g')
+            .attr('class', this.name+'_x_axis')
+            .attr('transform', 'translate('+this.margin.left+',' + (this.content_height+this.margin.top) + ')');
+        this.yAxis = this.svg.append('g')
+            .attr('class', this.name+'_y_axis')
+            .attr('transform', 'translate('+this.margin.left+',' + this.margin.top + ')');
+        this.option = option;
+        this.colorTheme = [d3.interpolateReds, d3.interpolateBlues] 
```

Comment:
Here, color seems to use to encode the height of a bar. Is this necessary?
But, it might be effective to easily distinguish different categories.

Reply:
Color encodes the category, such as top or bottom dynamic ranks; the warm color family encodes top ranks and the cool color theme encodes bottom ranks. Currently, the maximum number of the category is two, so I set two color family. Later it can be extended as the category grows.